File: python/Brunel/_note.py

__all__ = ["Note"]


def _setState(state, val, default=None):
    try:
        result = state[val]
        if result:
            return result
        else:
            return default
    except:
        return default


class Note:
    """Holds information about a Note in the network"""
    def __init__(self, props=None, getHook=None):
        self._getHook = getHook

        self.state = {
            "text": None,
            "id": None,
            "sources": [],
        }

        self.setState(props)

    def __str__(self):
        return f"Note({self.getText()})"

    def getID(self):
        return self.state["id"]

    def getText(self):
        return self.state["text"]

    def setState(self, state):
        if not state:
            return

        self.state["text"] = _setState(state, "text")
        self.state["id"] = _setState(state, "id")
        self.state["sources"] = _setState(state, "sources", [])

    def toDry(self):
        return {"value": self.state}

    @staticmethod
    def unDry(value):
        return Note(value)

    @staticmethod
    def load(data):
        return Note(data)

File: src/version7/Note.js

import Dry from "json-dry";

function setState(val, def=null){
  if (val){
    return val;
  } else {
    return def;
  }
}

class Note {
  constructor(props){
    this.state = {
      text: null,
      id: null,
      sources: [],
    };

    this.setState(props);

    this._getHook = null;
  }

  getID(){
    return this.state.id;
  }

  setState(state){
    if (state){
      this.state.text = setState(state.text);
      this.state.id = setState(state.id);
      this.state.sources = setState(state.sources, []);
    }
  }

  _updateHooks(hook){
    this._getHook = hook;
  }

  toString(){
    return `Note(${this.getText()})`;
  }

  getText(){
    return this.state.text;
  }

  toDry(){
    return {value: this.state};
  }
};

Note.unDry = function(value){
  return new Note(value);
}

Dry.registerClass(Note);

export default Note;
